### PR TITLE
 4-view-all-articles

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import ArticleList from "./article/ArticleList";
+
+const Home = () => {
+    return (
+        <div>
+            {/* <CreateTopicButton /> */}
```
Comment (MKD01) [inline src/components/Home.jsx:6]: Dont forget to remove commented out code.

File: src/components/article/ArticleList.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { useContext } from "react";
+import Article from "./Article";
+import { NewsDataContext } from "../../data/NewData";
+
+const ArticleList = () => {
+    const { articles, userList } = useContext(NewsDataContext);
```
Comment (MKD01) [inline src/components/article/ArticleList.jsx:6]: Is there a reason articles is within context here? Do you need access to articles anywhere else? Articles should really only need to exist within this component so it should not belong within your useContext.

File: src/data/NewData.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }, []);
+
+    useEffect(() => {
+        getAllUsers().then((users) => {
+            setUserList(users);
+        });
+    }, []);
+    useEffect(() => {
+        console.log(articleForComment);
+    }, [articleForComment]);
```
Comment (MKD01) [inline src/data/NewData.jsx:24]: Is this needed/doing anything?

File: src/data/NewData.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { getAllArticles, getAllUsers } from "../utils/api";
+
+export const NewsDataContext = createContext();
+
+export const NewsDataProvider = ({ children }) => {
+    const [articles, setArticles] = useState([]);
+    const [user, setUser] = useState("tickle122");
+    const [userList, setUserList] = useState([]);
+
+    const [articleForComment, setArticleForComment] = useState({});
```
Comment (MKD01) [inline src/data/NewData.jsx:11]: Should this be added in this ticket? Are you using this anywhere?

**Review by MKD01 — APPROVED** [review]
Good work so far, Im a big fan of the design and I like that its responsive as well 👍

However you have a lot of code/files here which should not belong on this ticket. Please only add code/files to that are relevant to the ticket that you are working on. I have also left a few comments for you to check out as well.

Once you have made all your changes im happy for you to merge this in.

File: src/data/NewData.jsx (hunk 3 above)
Comment (Hassen-Ahmed) [reply src/data/NewData.jsx:24]: not for now, but yes.

**Pull request closed without merging**